src/services/wallets/metamask/metamask.ts

```diff
@@ -0,0 +1,20 @@
+import Web3 from "web3";
+import { HexString } from "../../../utilities/types";
+import ethereum, { EthereumProvider } from "./ethereum";
+
+const mmweb3 = new Web3(ethereum as EthereumProvider);
```

Comment:
@Ajaxan have you tried this in a browser without metamask? I think it might blowup.
```suggestion
const mmweb3 = ethereum ? new Web3(ethereum) : null;
```

Reply:
No, it throws an error that is caught by the login sequence and display "Metamask not installed" in an error alert next to the login button.